### PR TITLE
KV access policy for VMSS, KV in different sub, improved docs, variable names cleaned up

### PR DESCRIPTION
**Pull request by dawg89**
Summary:

- ARM template resource added to give newly provisioned scale set access to read secrets from KV
- Terraform templates updated to support Key Vault in a different/remote subscription
- Improved docs/usability via updates to README.md, comments in Terraform templates, cleaning up names in templates

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -147,38 +153,73 @@ resource "azurerm_virtual_machine_scale_set" "scaleset" {
...
+            "${var.file2}"
           ]
         }
       SETTINGS
   }
 }
 
-
-resource "azurerm_key_vault" "kv" {
+# IMPLEMENTATION NOTE: The Terraform AzureRM provider does not currently support access policies
```
Comment (logachev) [inline terraform/main.tf]: I'm not sure we really need this workaround here. Would be great to update when Terraform adds support for this scenario.

File: terraform/main.tf [hunk 2]
```diff
@@ -147,38 +153,73 @@ resource "azurerm_virtual_machine_scale_set" "scaleset" {
     type_handler_version    = "2.0"
     settings                = <<SETTINGS
         {
-         "commandToExecute": "cmd",
+          "commandToExecute": "${var.command}",
           "fileUris": [
-            "https://<storagename>.blob.core.windows.net/<file>",
+            "${var.file1}",
```
Comment (logachev) [inline terraform/main.tf]: fileUris should be a variable, what if I add 3rd file?

Comment (dawg89) [reply terraform/main.tf]: I tried adding a single variable as a JSON array string, but it didn't work. The parser seems to only want to apply variables inside a set of quotes.

It's easy enough to add a third variable if needed, and the number of values would most likely be consistent across multiple environments.

**Comment by dawg89** [comment]
Responded

**Pull request merged**